aesd-char-driver/main.c

```diff
@@ -49,6 +50,102 @@ int aesd_release(struct inode *inode, struct file *filp)
     return 0;
 }
 
+//custom llseek function using fixed_size_llseek (Option 2)
+loff_t aesd_llseek( struct file *filp, loff_t offset, int whence )
+{
+    loff_t ret_offset;
+    struct aesd_dev *dev;
+    dev = filp->private_data;
+
+    // locking for the llseek
+    if (mutex_lock_interruptible(&aesd_device.lock)) {
+        ret_offset = -ERESTARTSYS;
+        goto end;
+    }
+
+    ret_offset = fixed_size_llseek(filp, offset, whence, dev->buffer.total_buff_size);
```

Comment:
@MaanasMDK how could you avoid locking during the `fixed_size_llseek` call here?